src/library.js

```diff
@@ -868,6 +868,35 @@ LibraryManager.library = {
     return -1;
   },
   pathconf: 'fpathconf',
+#if EMTERPRETIFY_ASYNC
+  fsync__deps: ['$FS', '__setErrNo', '$ERRNO_CODES', '$EmterpreterAsync'],
+  fsync: function(fildes) {
+    return EmterpreterAsync.handle(function(resume) {
+      // int fsync(int fildes);
+      // http://pubs.opengroup.org/onlinepubs/000095399/functions/fsync.html
+      var stream = FS.getStream(fildes);
+      if (stream) {
+        var mount = stream.node.mount;
+        if (!mount.type.syncfs) {
+          // We write directly to the file system, so there's nothing to do here.
+          resume(function() { return 0 });
```

Comment:
I don't think you can directly call `resume` in `handle`, which should trigger an error of bad state of EmterpreterAsync.
The best solution I have is to call it with `setTimeout`